src/worker/sandbox/cgroups/pool.go

```diff
@@ -23,20 +22,24 @@ type CgroupPool struct {
 	recycled chan *CgroupImpl
 	quit     chan chan bool
 	nextID   int
+	logger   slog.Logger
 }
 
 func NewCgroupPool(name string) (*CgroupPool, error) {
+
 	pool := &CgroupPool{
 		Name:     path.Base(path.Dir(common.Conf.Worker_dir)) + "-" + name,
 		ready:    make(chan *CgroupImpl, CGROUP_RESERVE),
 		recycled: make(chan *CgroupImpl, CGROUP_RESERVE),
 		quit:     make(chan chan bool),
 		nextID:   0,
+		logger:   common.CgTopLogger,
```

Comment:
why not just set it to what you want instead of immediately updating it after creating the struct?